test/replset_tests.js

```diff
@@ -2,9 +2,17 @@
 
 var co = require('co'),
   f = require('util').format,
-  assert = require('assert');
+  assert = require('assert'),
+  Promise = require('bluebird');
 
 describe('ReplSet', function() {
+  // Context variable stores all managers to clean up after test is completed
+  var managers = [];
```

Comment:
Just thought that it could be done easily enough with just a context variable. Very few lines are needed and I didn't think a class was necessary for this simple logic.

The developer still needs to know to use `newReplSet` rather than `new ReplSet`, which I think is as likely as forgetting `managers.push`.

Didn't consider the extensible part. With there being one `TestContext` class for the three tests, I'm not sure I can think of a  use case for setting `socketTimeout` or a similar option to a value for **every** test. The moment you want a test to not have that value, you can't use this approach anymore.

Would you like me switch to the `TestContext` approach?

Reply:
I was just thinking about how we got into this situation in the first place: it was too easy to forget to clean up something when a test failure occurred. To me it seemed like the best solution was "let's make a system where its really hard to _not_ clean up, in all cases", so the possibility of forgetting to add an instance to `managers` seems needlessly risky to me. Additionally, if you end up needing to change something you need to do it in three tests files. 

Having said that, I don't feel very strongly about this - we could easily do this refactor in the future if it turns out to be a problem.

Reply:
Yeah, I would say go ahead and merge this. If it becomes a problem like you said, it's an easy refactor.